x-pack/plugins/alerting/server/config.ts

```diff
@@ -8,6 +8,18 @@
 import { schema, TypeOf } from '@kbn/config-schema';
 import { validateDurationSchema } from './lib';
 
+const ruleTypeConfig = schema.object({
+  id: schema.string(),
+  actions: schema.maybe(schema.object({ max: schema.maybe(schema.number()) })),
```

Comment:
I see max actions can be configured by rule type, did we want that? I can see a future where we may want to configure by connector type (https://github.com/elastic/kibana/issues/126504) but not sure how complex it would get if we allow by rule types as well?

I’m thinking what we wanted by rule type was `timeout`?

Reply:
Actually this issue was for adding maxActions config globally and by ruleType:

```
We need to figure out the config structure for this so that we can provide a default, allow rule types to provide their own default and to give the user the ability to override this globally and per rule type
``` 

maxActions per connector type will be implemented with: https://github.com/elastic/kibana/issues/126504

And `timeout` will be implemented with: https://github.com/elastic/kibana/issues/124807
And yes `timeout` will be only in ruleType :)

maxActions: globally, by rule type and by connector type.
timeout: globally, by rule type, and by rule type registration...

i'll pick them up once this PR is merged :)

Reply:
Oh, I think this is my mistake for mixing my notes when typing that comment 🙈  I think I should have put that comment in https://github.com/elastic/kibana/issues/124807 instead 🤔 is it complicated to remove the possibility to set max actions by rule type? In my opinion, I think it would be better to leave the `actions` configuration agnostic from the rule types to keep things simple until we have a requirement to go further.

Reply:
Done, removed :)
If we want it under rule type again, can simply add back...
i left ruleType config there for now, will add `timeout there with https://github.com/elastic/kibana/issues/124807
an max actions by connector type with https://github.com/elastic/kibana/issues/126504